### ISSUE TITLE
Categorising problems

### ISSUE THREAD

**Issue by rross101** [human-substrate/Problems#4]
Looking at the existing problem list, would some categorisation / tagging be useful? E.g. fentanyl criris in Canada vs fentanyl crisis in USA - could there be 'global fentanyl crisis'.

Or - looking to where others have tried to do something similar - what about starting with the UN sustainable development goals - https://sdgs.un.org/goals - as potential categories?

**Comment by shahiryar** [comment]
Categorising problems is a good idea. SDGs, however, are too limiting when talking about the goal of substrate.
Also, SDG taxonomy will help in `Solutions` part.

**Comment by shahiryar** [comment]
We need a Substrate specific taxonomy.

**Comment by BryanKerrEdTech** [comment]
Just spitballing here, but do we risk premature optimization by creating a taxonomy this early? I wonder if we can leave the problem list completely unorganized and leverage gen AI to find what we need base on our requests/interests.

**Comment by shahiryar** [comment]
yeah... this is one way of starting. But here is what I think.
The problems already exist. Substrate is a place where they will be collected. The act of neat collection, in my view, requires an elegant taxonomy.
In fact, taxonomy of each of the repos might in itself be an issue/project per se.

However, to prevent over-optimization a rough classification could be used. 
Or even better **Keywords** could be used to initially.

**Comment by rross101** [comment]
I think we need some kind of database structure - for example, having location as a separate field.

**Comment by maxolasersquad** [comment]
Taxonomies always break down. e.g. "Is Johnny Cash country, rockabilly, rock, something else?" Things rarely fit into a single category. I believe a tagging system would be better. More concretely, global warming could be tagged with "health", "economy", "existential", etc.

**Comment by maxolasersquad** [comment]
Specifically, I'm thinking about the problem encountered in OOP with inheritance. We need to think more of [traits](https://en.wikipedia.org/wiki/Trait_(computer_programming)) instead of a hierarchical taxonomy.

**Comment by danielmiessler** [comment]
I agree with those saying we shouldn't classify. This is what AI is so good at, and it can do it dynamically whenever we want. Continuously improving. I'd say we punt on that for now.

**Comment by edu-ap** [comment]
An alternative would be to add tags such as #fentanyl #USA and #Canada in the relevant lines and then another file where relationships are mapped could relate #fentanyl to #opioid which then as a result would make it easy to map on a graph with a tool such as Obsidian (see [Graph View](https://help.obsidian.md/Plugins/Graph+view)). 

If `Problems` are stored in different files within the `Problems` directory then it would make it easy for these to be parsed by both humans and the AI agents that work with them. For example an user could have a `USA.md` and `Canada.md` files within the `Problems` directory based on their mental model if they chose to work that way.

**Comment by DesertEaglePWN** [comment]
I'm firmly in the camp of those who'd like to lossely organize via tags (I especially agree with [maxolasersquad](https://github.com/maxolasersquad))'s comparison to OOP inheritance. Don't think the organization is critical for reasons [danielmiessler](https://github.com/danielmiessler) mentioned, but it'll be helpful for access queries and profiling later on to have the data separate, even if the models can ultimately classify for cleaner presentation, manual tagging would allow for error correction and potentially RAG in the future. Not having any form of organizational oversight at all in the near future could also be detrimental if people onboard to the project pretty rapidly.

It may seem like a luxurious non-necessity, but nothing can truly scale without some structure in place. It may not be super relevant now, but it inevitably will be if the project grows.
